contrib/cindercompatibleapi/converter/volumetype.go

```diff
@@ -0,0 +1,342 @@
+// Copyright (c) 2018 Huawei Technologies Co., Ltd. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+This module implements a entry into the OpenSDS northbound service.
+*/
+
+package converter
+
+import (
+	"errors"
+
+	"github.com/opensds/opensds/pkg/model"
+)
+
+// *******************Create a volume type*******************
+
+// CreateTypeReqSpec ...
+type CreateTypeReqSpec struct {
+	VolumeType CreateReqVolumeType `json:"volume_type,omitempty"`
+}
+
+// CreateReqVolumeType ...
+type CreateReqVolumeType struct {
+	Name           string    `json:"name"`
+	AccessIsPublic bool      `json:"os-volume-type-access:is_public,omitempty"`
+	Description    string    `json:"description,omitempty"`
+	Extras         ExtraSpec `json:"extra_specs,omitempty"`
+}
+
+// CreateTypeRespSpec ...
+type CreateTypeRespSpec struct {
+	VolumeType CreateRespVolumeType `json:"volume_type,omitempty"`
+}
+
+// CreateRespVolumeType ...
+type CreateRespVolumeType struct {
+	IsPublic       bool      `json:"is_public,omitempty"`
+	Extras         ExtraSpec `json:"extra_specs,omitempty"`
+	Description    string    `json:"description,omitempty"`
+	Name           string    `json:"name,omitempty"`
+	ID             string    `json:"id,omitempty"`
+	AccessIsPublic bool      `json:"os-volume-type-access:is_public,omitempty"`
+}
+
+// CreateTypeReq ...
+func CreateTypeReq(cinderReq *CreateTypeReqSpec) (*model.ProfileSpec, error) {
```

Comment:
I don't think this method should return error, since other methods only return ***RespSpec. You can package the validate operation in a standalone method.